hnn_core/tests/test_mpi_child.py

```diff
@@ -56,10 +131,19 @@ def test_empty_data():
 
 
 def test_data_len_mismatch():
-    """Test that an unexpected data length raises RuntimeError"""
-    data_bytes = b'\0'
-    expected_len = 2
+    """Test that padded data can be unpickled with warning for length """
+
+    with MPISimulation(skip_mpi_import=True) as mpi_sim:
+        pickled_bytes = mpi_sim._pickle_data({})
+
+    expected_len = len(pickled_bytes) + 1
```

Comment:
I didn't notice it before but why the + 1 ?

Reply:
this triggered the UserWarning, but it may not be necessary when base64 encoding is removed from this PR